aviary/subsystems/propulsion/engine_deck.py

```diff
@@ -973,15 +978,36 @@ def build_mission(self, num_nodes, aviary_inputs) -> om.Group:
             engine_group.add_subsystem(
                 'max_thrust_interpolation',
                 max_thrust_engine,
-                promotes_inputs=['*'],
-                promotes_outputs=['*'])
+                promotes_inputs=['*'])
 
         engine_group.add_subsystem('engine_scaling',
                                    subsys=EngineScaling(num_nodes=num_nodes,
                                                         aviary_options=self.options),
-                                   promotes_inputs=['*'],
+                                   promotes_inputs=[
+                                       Aircraft.Engine.SCALE_FACTOR, Dynamic.Mission.MACH],
                                    promotes_outputs=['*'])
 
+        # manually connect unscaled variables, since we do not want them promoted
+        engine_group.connect('interpolation.thrust_net_unscaled',
+                             'engine_scaling.thrust_net_unscaled')
+        engine_group.connect('interpolation.fuel_flow_rate_unscaled',
+                             'engine_scaling.fuel_flow_rate_unscaled')
+        engine_group.connect('interpolation.electric_power_unscaled',
+                             'engine_scaling.electric_power_unscaled')
+        engine_group.connect('interpolation.nox_rate_unscaled',
+                             'engine_scaling.nox_rate_unscaled')
+        if self.use_thrust:
+            engine_group.connect(
+                'max_thrust_interpolation.thrust_net_max_unscaled', 'engine_scaling.thrust_net_max_unscaled')
+
+        if self.use_shaft_power:
+            if SHAFT_POWER in self.engine_variables:
+                engine_group.connect('interpolation.shaft_power_unscaled',
+                                     'engine_scaling.shaft_power_unscaled')
+            else:
```

Comment:
the `use_shaft_power` flag only exists if one of the shaft power variables already exists in the engine deck, so that check has basically already happened